crates/forge/tests/e2e/docs_snippets_validation.rs

```diff
@@ -1,43 +1,91 @@
+            return true;
+        }
+    }
+    false
+}
+
 #[test]
 fn test_docs_snippets() {
+    let listings_to_packages_mapping = create_listings_to_packages_mapping();
+
     let root_dir = get_parent_dir(2);
     let docs_dir = root_dir.join("docs/src");
 
-    let re = Regex::new(r"(?ms)```shell\n\$ (snforge .+?)\n```").expect("Invalid regex pattern");
-
-    let snippets = extract_snippets_from_directory(&docs_dir, &re)
-        .expect("Failed to extract snforge command snippets");
+    let snippet_type = SnippetType::Forge;
 
-    // TODO(#2684)
-    let skipped_args = [
-        // for some reason `try_parse_from` fails on `--version` flag
-        vec!["snforge", "--version"],
-    ];
+    let snippets = extract_snippets_from_directory(&docs_dir, &snippet_type)
+        .expect("Failed to extract command snippets");
 
     for snippet in &snippets {
         let args = snippet.to_command_args();
-        let args: Vec<&str> = args.iter().map(String::as_str).collect();
+        let mut args: Vec<&str> = args.iter().map(String::as_str).collect();
 
-        if skipped_args.contains(&args) {
-            print_skipped_snippet_message(snippet, "snforge");
+        if snippet.ignored {
+            print_skipped_snippet_message(snippet);
             continue;
         }
 
-        let parse_result = Cli::try_parse_from(args);
+        let parse_result = Cli::try_parse_from(args.clone());
         let err_message = if let Err(err) = &parse_result {
             err.to_string()
         } else {
             String::new()
         };
-        assert_valid_snippet(parse_result.is_ok(), snippet, "snforge", &err_message);
+
+        assert_valid_snippet(parse_result.is_ok(), snippet, &err_message);
+
+        // Remove "snforge" from the args
+        args.remove(0);
+
+        // Remove "test" from the args
+        args.retain(|element| element != &"test");
+
+        if let Some(snippet_output) = &snippet.output {
+            let package_name = snippet
+                .capture_package_from_output()
+                .expect("Failed to capture package from command output");
+
+            // TODO(#2698)
+            let temp = resolve_temp_dir(&package_name, &listings_to_packages_mapping);
+            let output = test_runner(&temp).args(args).assert();
+
+            assert_stdout_contains(output, snippet_output);
+        }
     }
 
-    print_success_message(snippets.len(), "snforge");
+    print_success_message(snippets.len(), snippet_type.as_str());
+}
+
+fn resolve_temp_dir(
+    package_name: &str,
+    listings_to_packages_mapping: &HashMap<String, Vec<String>>,
+) -> TempDir {
+    if is_package_from_docs_listings(package_name, listings_to_packages_mapping) {
+        setup_package_from_docs_listings(package_name, listings_to_packages_mapping)
+    } else if ["addition", "fibonacci"].contains(&package_name) {
+        setup_hello_workspace()
+    } else {
+        setup_package(package_name)
+    }
 }
```

Comment:
note: Currently we use `hello_workpaces` package (from `crates/forge/tests/data`) in one example in docs. In future we can simplify this logic and remove `is_package_from_docs_listings`, once we use only packages from docs listings. Ref: https://github.com/foundry-rs/starknet-foundry/issues/2698

Reply:
No longer relevant.